Demo/Common/IntegrationExamples/Components/CardComponentExample.swift

```diff
@@ -12,50 +12,67 @@ import AdyenSession
 internal final class CardComponentExample: InitialDataFlowProtocol {
 
     // MARK: - Properties
-
-    internal var session: AdyenSession?
-    internal var cardComponent: PresentableComponent?
+    
     internal weak var presenter: PresenterExampleProtocol?
 
+    private var session: AdyenSession?
+    private var cardComponent: PresentableComponent?
+
     // MARK: - Initializers
 
     internal init() {}
 
-    // MARK: - Networking
-
-    internal func requestInitialData(completion: ((PaymentMethods?, Error?) -> Void)?) {
-        requestAdyenSessionConfiguration { [weak self] adyenSessionConfig, errorResponse in
-            guard let self = self else {
-                return
-            }
-            guard let config = adyenSessionConfig else {
-                return
-            }
-            AdyenSession.initialize(with: config,
-                                    delegate: self,
-                                    presentationDelegate: self) { [weak self] result in
-                switch result {
+    internal func present() {
+        presenter?.showLoadingIndicator()
+        requestInitialData { [weak self] response in
+            self?.presenter?.hideLoadingIndicator { [weak self] in
+                
+                guard let self else { return }
+                
+                switch response {
                 case let .success(session):
-                    self?.session = session
-                case let .failure(errorResponse):
-                    self?.presentAlert(with: errorResponse)
+                    guard let component = self.cardComponent(from: session.sessionContext.paymentMethods) else {
```

Comment:
Same as above, splitting the present and load

Reply:
d6e580c29b4b8c6d9048e53bfc295488ddf025a6